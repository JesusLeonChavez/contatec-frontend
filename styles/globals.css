:root{
  --primary: #706F6F;
  --secondary: #304659;
}

html,
body {
  padding: 0;
  margin: 0;

}

a {
  color: inherit;
  text-decoration: none;
}

* {
  box-sizing: border-box;
  font-family: 'Poppins', sans-serif;
}

.generalWrapper{
  max-width: 1300px;
  margin: 0 auto;
  overflow: auto;
  padding: 0 40px;
}

.flexCenterCustom{
  display: flex;
  align-items: center;
  justify-content: center;
}


/*----------------Swiper-----------------------*/
.swiper-container {
  width: 100%;
  height: 100%;
}

.swiper-slide {
  text-align: center;
  font-size: 18px;
  background: red;
  /* Center slide text vertically */
  display: -webkit-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-align-items: center;
  align-items: center;
}

.slide-2{
  background: green;
}

.mySwiper  .swiper-button-next, .mySwiper  .swiper-button-prev {
  color: white;
  border: 1px solid white;
  border-radius: 50%;
  padding: 1.3rem;
}

.mySwiper .swiper-button-next{
  top: 800px;
  right: 100px;
}

.mySwiper .swiper-button-prev{
  top: 800px;
  right: 180px;
  left: auto;
}

.mySwiper .swiper-button-next::after,.mySwiper .swiper-button-prev::after{
  font-size: 1rem;
}


.mySwiper2{
  width: 200px;
}
.mySwiper2 .swiper-container{
  width: 100%;
  height: 100%;
}
.mySwiper2 .swiper-slide{
  text-align: center;
  font-size: 18px;
  background: #fff;
  background: pink;
  /* Center slide text vertically */
  display: -webkit-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-align-items: center;
  align-items: center;
}
.mySwiper2 .slide-2{
  background: blue;
}


.mySwiper3{
  width:100%;
}
.mySwiper3 .swiper-container{
  width: 300px;
  height: 100%;
}
.mySwiper3 .swiper-slide{
  text-align: center;
  font-size: 18px;
  background: #fff;
  background: purple;
  height: 500px;
  /* Center slide text vertically */
  display: -webkit-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-align-items: center;
  align-items: center;
}
.mySwiper3 .slide-2{
  background: blue;
}

.mySwiper3 .swiper-button-next, .mySwiper3 .swiper-button-prev {
  color: white;
  border: 1px solid white;
  border-radius: 50%;
  padding: 1.3rem;
}

.mySwiper3 .swiper-button-next{
  top: 450px;
  right: 100px;
}
.mySwiper3 .swiper-button-prev{
  top: 450px;
  right: 180px;
  left: auto;
}

.mySwiper3 .swiper-button-next::after, .mySwiper3 .swiper-button-prev::after{
  font-size: 1rem;
}




