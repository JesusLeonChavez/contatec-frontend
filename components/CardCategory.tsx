/* eslint-disable camelcase */
import { Box, Text, Flex } from "@chakra-ui/react"
import ZIcon from "../components/Icon/ZIcon"

import styles from "../styles/sections/Home.module.css"
import Image from "next/image"
import ModalNewPost from "../sections/Post/ModalNewPost"

interface User {
  avatar: string
  createdAt: string
  id: number
  updatedAt: string
  us_apellido: string
  us_correo: string
  us_nombre: string
}

interface Post {
  id: number
  createdAt: string
  updatedAt: string
  pst_isActive: boolean
  pst_descripcion_corta: string
  pst_nombre: string
  pst_descripcion_incluye: string
  pst_descripcion: string
  pst_imagen_1: string
  pst_imagen_2: string
  pst_imagen_3: string
  pst_imagen_4: string
  pst_imagen_5: string
  pst_precioBase: number
  pstUsuarioId: User
}

interface PropsCard {
  post: Post
  categoryScreen?: boolean
}
export default function CardCategory({
  post,
  categoryScreen = true
}: PropsCard) {
  return (
    <Box borderRadius="lg" overflow="hidden" mx="3">
      <Box position="relative">
        <Image
          src={post?.pst_imagen_5 || "/assets/images/marketing/marketing1.png"}
          alt={post?.pst_nombre}
          height="500"
          width="500"
        />
        <Flex
          align="center"
          justify="space-between"
          w="full"
          p="3"
          position="absolute"
          bottom="1"
          backgroundColor="whiteAlpha.700"
        >
          <Flex align="center" w="40px" justify="space-between">
            <ZIcon name="star" color="secondary" />
            <Text fontSize="sm" className={styles.bold400} color="primary">
              4.0
            </Text>
          </Flex>
          {categoryScreen ? (
            <Text fontSize="sm" className={styles.bold400} color="primary">
              S/. {post?.pst_precioBase}
            </Text>
          ) : (
            <ModalNewPost variant="primary" width="" icon mypost={post} />
          )}
        </Flex>
      </Box>

      <Box px="2" pb="5">
        <Flex align="flex-start" justify="center" direction="column">
          {/* <Flex align="center" justify="space-between" w="full" p="3">
            <Flex align="center" w="40px" justify="space-between">
              <ZIcon name="star" />
              <Text fontSize="sm" className={styles.bold200} color="primary">
                4.0
              </Text>
            </Flex>
            {categoryScreen ? (
              <Text fontSize="sm" className={styles.bold200} color="primary">
                S/. {post?.pst_precioBase}
              </Text>
            ) : (
              <ModalNewPost variant="primary" width="sm" icon />
            )}
          </Flex> */}
          <Flex align="center" justify="flex-start">
            <Text fontSize="md" className={styles.bold500} color="primary">
              {post?.pst_nombre}
            </Text>
          </Flex>
          {categoryScreen && (
            <Flex align="center" justify="flex-start">
              <Text fontSize="sm" className={styles.bold200} color="primary">
                Por {post.pstUsuarioId.us_nombre}
                {post.pstUsuarioId.us_apellido}
              </Text>
            </Flex>
          )}
        </Flex>
      </Box>
    </Box>
  )
}
