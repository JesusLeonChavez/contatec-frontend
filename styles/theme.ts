import { extendTheme } from '@chakra-ui/react'
import { ButtonStyles as Button } from './components/buttonStyles'
export const newTheme = extendTheme({
  colors: {
    primary: '#EB0029',
    secondary: '#FF6F91',
    highlight: '#00C9A7'
  },
  components: {
    Button
  }
})
