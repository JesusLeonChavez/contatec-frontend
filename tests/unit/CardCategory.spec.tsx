/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react"

import "@testing-library/jest-dom"

import { newTheme } from "../../styles/theme"

import { ChakraProvider } from "@chakra-ui/react"

import CardCategory from "../../components/CardCategory"

const ChakraRenderer = ({ children }) => {
  return <ChakraProvider theme={newTheme}>{children}</ChakraProvider>
}

describe("CardCategory", function () {
  let expectedProps

  beforeEach(() => {
    expectedProps = {
      title: "New Bar",
      imageUrl: "/assets/marketing/marketing1.png"
    }
  })

  test("debe renderizarse el componente", () => {
    const component = render(<CardCategory {...expectedProps} />, {
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      wrapper: ChakraRenderer
    })
    expect(component).toMatchSnapshot()
  })

  test("debe recibir las propiedades con el tipo esperado", () => {
    expect(typeof expectedProps.title).toBe("string")
    expect(typeof expectedProps.imageUrl).toBe("string")
  })

  test("debe renderizar title y image", () => {
    const { getByText, getByAltText } = render(
      <CardCategory {...expectedProps} />,
      {
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-ignore
        wrapper: ChakraRenderer
      }
    )

    const title = getByText(expectedProps.title)
    const img = getByAltText(expectedProps.title)

    expect(title).toBeVisible()
    expect(img).toBeVisible()
  })
})
