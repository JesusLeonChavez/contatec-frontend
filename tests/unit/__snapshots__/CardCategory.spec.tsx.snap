// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardCategory debe renderizarse el componente 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-1oqqhbx"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTAwIiBoZWlnaHQ9IjUwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            alt="New Bar"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
        <div
          class="css-kh5fs6"
        >
          <div
            class="css-18f4n7l"
          >
            <div
              class="css-qf8moh"
            >
              <div
                class="css-r5paij"
              >
                <svg
                  color="var(--primary)"
                  fill="none"
                  height="1em"
                  style="font-size: 15px;"
                  viewBox="0 0 20 20"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M3.5 8.695l4.547-.702L9.998 3.8l1.952 4.193 4.55.702-3.3 3.418.755 4.687-3.957-2.197L6.043 16.8l.754-4.687L3.5 8.695z"
                    fill="var(--primary)"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M10 15.925L4.438 19 5.5 12.488 1 7.875l6.219-.95L10 1l2.781 5.925 6.219.95-4.5 4.613L15.562 19 10 15.925zM4.12 8.92l2.984 3.063-.683 4.199L10 14.214l3.58 1.968-.683-4.2 2.985-3.062-4.116-.63L10 4.534 8.234 8.29l-4.113.63z"
                    fill="#FBD76D"
                    fill-rule="evenodd"
                  />
                </svg>
                <p
                  class="chakra-text bold200 css-3hfsa6"
                >
                  4.0
                </p>
              </div>
              <p
                class="chakra-text bold200 css-3hfsa6"
              >
                S/. 200
              </p>
            </div>
            <div
              class="css-geek62"
            >
              <p
                class="chakra-text bold500 css-11pzpln"
              >
                New Bar
              </p>
            </div>
            <div
              class="css-geek62"
            >
              <p
                class="chakra-text bold200 css-3hfsa6"
              >
                Por Luis Sotelo
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1oqqhbx"
    >
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTAwIiBoZWlnaHQ9IjUwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
            style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
          />
        </div>
        <noscript />
        <img
          alt="New Bar"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
      <div
        class="css-kh5fs6"
      >
        <div
          class="css-18f4n7l"
        >
          <div
            class="css-qf8moh"
          >
            <div
              class="css-r5paij"
            >
              <svg
                color="var(--primary)"
                fill="none"
                height="1em"
                style="font-size: 15px;"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M3.5 8.695l4.547-.702L9.998 3.8l1.952 4.193 4.55.702-3.3 3.418.755 4.687-3.957-2.197L6.043 16.8l.754-4.687L3.5 8.695z"
                  fill="var(--primary)"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M10 15.925L4.438 19 5.5 12.488 1 7.875l6.219-.95L10 1l2.781 5.925 6.219.95-4.5 4.613L15.562 19 10 15.925zM4.12 8.92l2.984 3.063-.683 4.199L10 14.214l3.58 1.968-.683-4.2 2.985-3.062-4.116-.63L10 4.534 8.234 8.29l-4.113.63z"
                  fill="#FBD76D"
                  fill-rule="evenodd"
                />
              </svg>
              <p
                class="chakra-text bold200 css-3hfsa6"
              >
                4.0
              </p>
            </div>
            <p
              class="chakra-text bold200 css-3hfsa6"
            >
              S/. 200
            </p>
          </div>
          <div
            class="css-geek62"
          >
            <p
              class="chakra-text bold500 css-11pzpln"
            >
              New Bar
            </p>
          </div>
          <div
            class="css-geek62"
          >
            <p
              class="chakra-text bold200 css-3hfsa6"
            >
              Por Luis Sotelo
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
